### PR TITLE
[TDI-25] [TDI-29] [TDI-36] Add ability to create columns and cards

### PR DESCRIPTION
**Pull request by sebxian**
## Description

This PR...

* Introduces the `type: "page"` attribute as required on the root level of the page object. This greatly simplifies recursive functions parsing the structure of the file (but will break existing pages missing the attribute).
* Improves the existing dialogs for text and headline blocks, closing TDI-25.
* Adds the ability to add columns and card (will appear inline, without the need to open a dialog).

### Notes for testing:

* Create a blank page and add blocks, nesting blocks in blocks (some combinations should be prevented, e.g., columns inside columns, cards inside cards).
* Also test editing and deleting blocks, making sure none of the previous functions have been broken.

## Checklist

- [x] Documentation updated if required
- [x] Test coverage is appropriate
      
## Checklist Internal

- [x] Linear issue linked (e.g. [DIR-XXXX] pull request title)
- [x] Has the PR been labeled

### CODE REVIEW

File: ui/src/pages/namespace/Explorer/Page/poc/schema/blocks/index.ts [hunk 1]
```diff
@@ -55,3 +55,6 @@ export const AllBlocks: z.ZodType<AllBlocksType> = z.lazy(() =>
 export const TriggerBlocks = z.discriminatedUnion("type", [Button]);
 
 export type TriggerBlocksType = z.infer<typeof TriggerBlocks>;
+
+/* Inline blocks do not need a dialog for creation */
+export const inlineBlockTypes: AllBlocksType["type"][] = ["columns", "card"];
```
Comment (stefan-kracht) [inline ui/src/pages/namespace/Explorer/Page/poc/schema/blocks/index.ts:60]: There is a 90 % chance that I would have written this the exact same way, but when I think about it, shouldn't this be a Set instead of an Array?

File: ui/src/pages/namespace/Explorer/Page/poc/PageCompiler/context/utils/index.ts [hunk 2]
```diff
@@ -126,7 +128,30 @@ export const pathsEqual = (a: PathOrNull, b: PathOrNull) => {
...
 
-export const getPlaceholderBlock = (type: AllBlocksType["type"]) => {
+type ParseAncestorsConfig = {
+  page: DirektivPagesType;
+  path: BlockPathType;
+  fn: (block: AllBlocksType | DirektivPagesType) => boolean;
+  depth?: number;
+};
+
+export const parseAncestors = ({
```
Comment (stefan-kracht) [inline ui/src/pages/namespace/Explorer/Page/poc/PageCompiler/context/utils/index.ts:138]: Very good implementation (and test coverage) of that helper function. I was just wondering if the name is a bit too generic. What do you think about renaming it to something like 

```
findAncestors({
  page,
  path,
  match: (block) => block.type === "columns",
}),
```

File: ui/src/pages/namespace/Explorer/Page/poc/PageCompiler/context/pageCompilerContext.tsx [hunk 3]
```diff
@@ -115,4 +122,25 @@ export const usePageEditor = () => {
...
+      return addBlock(path, getBlockTemplate(type), true);
+    }
+    setDialog({
+      action: "create",
+      block: getBlockTemplate(type),
+      path,
+    });
+  };
+
+  return { createBlock };
```
Comment (stefan-kracht) [inline ui/src/pages/namespace/Explorer/Page/poc/PageCompiler/context/pageCompilerContext.tsx:145]: What do think about only returning a function here?

File: ui/src/pages/namespace/Explorer/Page/poc/PageCompiler/context/pageCompilerContext.tsx [hunk 4]
```diff
@@ -115,4 +122,25 @@ export const usePageEditor = () => {
   };
 };
 
-export { PageCompilerContextProvider };
+/**
+ * This hook returns createBlock(), which opens the editor dialog for
+ * blocks such as text blocks, or just adds an inline block to the page
+ * (e.g., cards or columns, where no dialog is required).
+ */
+export const useCreateBlock = () => {
```
Comment (stefan-kracht) [inline ui/src/pages/namespace/Explorer/Page/poc/PageCompiler/context/pageCompilerContext.tsx:130]: I wonder if this hook belongs into the `pageCompilerContext.tsx` file? Would you mind placing it next to the 
`useBlockTypes.tsx` in the `utils` directory?

**Review by stefan-kracht — CHANGES_REQUESTED** [review]
Very nice!! I made some suggestions in the comments (all optional) but I also found  found a small bug:

The user can nest a card inside a card when it is the root:

https://github.com/user-attachments/assets/ebad6586-5e16-4776-bea6-4e434db4d371

File: ui/src/pages/namespace/Explorer/Page/poc/PageCompiler/context/pageCompilerContext.tsx (hunk 4 above)
Comment (sebxian) [reply ui/src/pages/namespace/Explorer/Page/poc/PageCompiler/context/pageCompilerContext.tsx]: Good idea, fixed in 013602a65150ddbafa64bb8fbf8480b1eb6901d8
